### ISSUE TITLE
Pattern with a "one-or-more" quantifier not working as expected

### ISSUE THREAD

**Issue by FenstonSingel** [Genivia/RE-flex#92]
Program in question:
```
%%
-?[0-9]+    out() << '[' << text() << ']';
%%
int main() { return Lexer().lex(); }
```

Test input:
```
4 -8 - -
```

Expected output:
```
[4] [-8] - -
```

Output:
```
[4] [-8] [-] [-]
```

Translated into a cpp-program with `reflex bug.l`, compiled with `g++ lex.yy.cpp -lreflex -o bug`.
RE/flex version 2.1.5, g++ version 10.2.0.

Also attaching a resulting state machine acquired with `--graphs-file` flag:
![image](https://user-images.githubusercontent.com/21692591/95675969-9f0eaf00-0bc3-11eb-8e9a-9af20e7beb12.png)

**Comment by genivia-inc** [comment]
Looks like a problem indeed. Thanks for reporting this!

This is unusual, as we do a lot of testing. Nevertheless, bugs cannot be 100% eradicated.

Interestingly, `-?[0-9][0-9]*` works correctly and also `-?[0-9]{1,}` works correctly. Both match the same pattern. So something in the so-called subset construction algorithm is off.

**Comment by genivia-inc** [comment]
I found the problem, which is very localized to a specific configuration of the generated DFA. There is an optimization step in the DFA subset construction algorithm to optimize the DFA moves from the follow set. This optimization step was changed in RE/flex v2.0. to optimize this step to increased its speed, but unfortunately the change caused one error on one line. An update will be released soon. This update also includes several improvements related to the RE/flex regex engine for high-performance searching.
